packages/data-stores/src/auth/actions.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Internal dependencies
+ */
+import {
+	AuthOptionsSuccessResponse,
+	AuthOptionsErrorResponse,
+	WpLoginSuccessResponse,
+	WpLoginErrorResponse,
+} from './types';
+
+export const reset = () =>
+	( {
+		type: 'RESET_LOGIN_FLOW' as const,
+	} as const );
+
+export const receiveAuthOptions = (
+	response: AuthOptionsSuccessResponse,
+	usernameOrEmail: string
+) =>
+	( {
+		type: 'RECEIVE_AUTH_OPTIONS',
+		response,
+		usernameOrEmail,
+	} as const );
+
+export const receiveAuthOptionsFailed = ( response: AuthOptionsErrorResponse ) =>
+	( {
+		type: 'RECEIVE_AUTH_OPTIONS_FAILED',
+		response,
+	} as const );
+
+export interface FetchAuthOptionsAction {
+	type: 'FETCH_AUTH_OPTIONS';
+	usernameOrEmail: string;
+}
+
+const fetchAuthOptions = ( usernameOrEmail: string ): FetchAuthOptionsAction => ( {
+	type: 'FETCH_AUTH_OPTIONS',
+	usernameOrEmail,
+} );
+
+export function* submitUsernameOrEmail( usernameOrEmail: string ) {
+	try {
+		const authOptions = yield fetchAuthOptions( usernameOrEmail );
+
+		yield receiveAuthOptions( authOptions, usernameOrEmail );
+	} catch ( err ) {
+		yield receiveAuthOptionsFailed( err );
+	}
+}
+
+export const receiveWpLogin = ( response: WpLoginSuccessResponse ) =>
+	( {
+		type: 'RECEIVE_WP_LOGIN',
+		response,
+	} as const );
+
+export const receiveWpLoginFailed = ( response: WpLoginErrorResponse ) =>
+	( {
+		type: 'RECEIVE_WP_LOGIN_FAILED',
+		response,
+	} as const );
+
+type WpLoginAction = 'login-endpoint' | 'two-step-authentication-endpoint';
+
+export interface FetchWpLoginAction {
+	type: 'FETCH_WP_LOGIN';
+	action: WpLoginAction;
+	params: object;
+}
+
+const fetchWpLogin = ( action: WpLoginAction, params: object ): FetchWpLoginAction =>
+	( {
+		type: 'FETCH_WP_LOGIN',
+		action,
+		params,
+	} as const );
+
+export function* submitPassword( password: string ) {
+	const username = yield { type: 'SELECT_USERNAME_OR_EMAIL' };
+
+	try {
+		const loginResponse = yield fetchWpLogin( 'login-endpoint', { username, password } );
```

Comment:
This returns a JSON blob even if a login fails (e.g. incorrect password with code `incorrect_password`), so the below `receiveWpLogin` call means that when I run `wp.auth.submitPassword( 'incorrect-passw0rd' );` in the console, I get `New loginFlowState = LOGGED_IN`, where it should error out.

Reply:
Hmmm, yeah was obviously expecting a throw here, but you're right my intention was to do error cases et al. in subsequent PRs.

Cheers all!